migration/src/spreadsheets.rs

```diff
@@ -130,6 +132,68 @@ impl SheetResult {
         })
     }
 
+    /// Parse this sheet as the collection index. Updates a Collection 
+    /// which contains two Chapter Groups, each of which 
+    /// has a list of associated Chapters.
+    pub fn into_collection_index(self,
+        self_title: &String, 
+        self_wordpress_menu_id: &i64) -> Result<Collection> {
+        let mut chapters = Vec::new();
+        let mut self_intro_chapters = Vec::new();
+        for row in self.values.into_iter().skip(2){
+            if row[0].is_empty(){
+                self_intro_chapters = chapters.clone();
+                chapters = Vec::<Chapter>::new();
+            }
+            else {
+                let index_in_parent_string = row[0].clone();
+                let index_i64 = index_in_parent_string.parse::<i64>().unwrap();
+
+                let document_if_applicable = row[3].clone();
+                let mut doc_string= None;
+
+                let wp_id_if_applicable = row[5].clone();
+                let mut wp_id = None;
+
+                if wp_id_if_applicable.is_empty(){
+                    wp_id = None;
+                }
+                else {
+                    wp_id = Some(wp_id_if_applicable.parse::<i64>().unwrap());
+                }
+
+                if document_if_applicable.is_empty() {
+                    doc_string = None;
+                }
+                else {
+                    doc_string = Some(document_if_applicable);
+                }
+
+                let new_chapter = Chapter {
+                    index_in_parent: index_i64,
+                    url_slug: row[1].clone(),
+                    chapter_name: row[2].clone(),
+                    document_title: doc_string,
+                    id: None,
+                    document_id: None,
```

Comment:
??? You'll need the document's `short_name` field here, which perhaps I should have called slug. But anyway, so you could call it `document_short_name: String` perhaps since you don't have the document's `id` yet (`DocumentId` is a wrapper for UUID) and we don't need the `title` field.

Reply:
Oh, okay-- I'll change the Chapter struct to reflect this, and make the Document Title field optional on the spreadsheet, and the Document ID field required (if applicable) on the spreadsheet to reflect this as well.

Reply:
Fixed so now the short name is used to get the document ID instead of the document title.